required/latex-lab/latex-lab-graphic.dtx

```diff
@@ -137,37 +388,211 @@
 %   responsibility of the surrounding code to add appropriate tagging commands.
-%   \item[\meta{name}] Other values will be used as tag names in the structure. If the tag is not
-%   known as a structure tag you will get an warning from tagpdf. The default name is currently
-%   \texttt{Figure}
+%   \item[\texttt{tag=\meta{name}}] This switches to the illustrative mode but uses
+%   \meta{name} as tag name in the structure instead of the default \texttt{Figure}. 
+%   This can for example be used to tag an image of a formula with \texttt{Formula}.
 %   \end{description}   
-%  \item[\texttt{actualtext}] This allows to add an /ActualText to the structure. 
-%  This is useful for small graphics that represent single chars or a short word like a 
-%  logo. If \texttt{actualtext} is used, the graphics is not enclosed in \texttt{Figure}
-%  structure but in a \texttt{Span} structure and no \texttt{/BBox} attribute is added.
-%  This in accordance with (the draft of) PDF/UA-2 but violates perhaps PDF/UA-1.
 %  \item[\texttt{correct-BBox}] If the calculated \texttt{/BBox} values are wrong
-%  they can be correct with this key. It expects four dimensions that are added to the 
+%  they can be corrected with this key. It expects four dimensions that are added to the 
 %  \texttt{/BBox} values. 
-%  \item[\texttt{debug}] The value \texttt{BBox} will show the calculated  \texttt{/BBox} 
-%  as a half transparent red rectangle. 
 %  \end{description}
 %  
-%  The code also redefines the \texttt{alt} key to actually add its values
-%  as an alternative text. If no \texttt{alt} value is given, a warning is issued 
-%  and the file name of the graphic is used.
+%  The code also add to the \texttt{debug} key of \cs{DocumentMetadata} the value \texttt{BBox}.
+%  This adds a half transparent red layer showing the calculated BBox.
+%  The key can only be used after that graphic module has been loaded, which means
+%  one needs a second \cs{DocumentMetadata} call:
+%  \begin{verbatim}
+%   \DocumentMetadata{tagging=on}
+%   \DocumentMetadata{debug=BBox}
+%  \end{verbatim}
+%  
+%  \section{Make \env{picture} tagging aware}
+%  
+%  \subsection{User interface}
+%  The \env{picture} environment doesn't have a optional argument with key-value value processing.
+%  So the code changes that and then provides the keys already discussed for \cs{includegraphics}.
+%  
+%  \subsection{BBox calculation}
+%  
+%  The BBox calculation is much simpler than for \cs{includegraphics} as the code simply takes
+%  the declared size from the \env{picture} arguments. 
 %  
+%  \subsection{Tagging in text mode}
+%  
+%  A  \env{picture} can contain \cs{put} commands with text content, tagging this in 
+%  text mode could make sense in some cases. But it is not quite clear if one can/should 
+%  redefine the \cs{put} command. If text tagging is wanted we suggest to 
+%  define a dedicated command along these lines:
+%  \begin{verbatim}
+%  \NewDocumentCommand\picturenode{O{}r()m}
```

Comment:
`\picturenode`   couldn't this be `\put{tagging-setup={text}](..){...}`  or some such?